### PR TITLE
[ET-858] Clean up RSVP input names

### PR DESCRIPTION
**Pull request by sc0ttkclark**
[ET-858]

This PR cleans up the various input names for the new RSVP FE views. This approach is backwards compatible so that old inputs and old views can still be used.

Working on tests for this.

## Old POST data

```php
$old_post_data = [
	'product_id'             => [
		$ticket_id,
	],
	'quantity_' . $ticket_id => 2,
	'attendee'               => [
		'order_status' => 'yes',
		'full_name'    => 'Me Who',
		'email'        => 'me@who.com',
	],
	'tribe-tickets-meta'     => [
		$ticket_id => [
			0 => [
				$field_slug => 'Attendee 1 field value',
			],
			1 => [
				$field_slug => 'Attendee 2 field value',
			],
		],
	],
];
```

## New POST data

```php
$new_post_data = [
	'tribe_tickets' => [
		'ticket_id' => $ticket_id,
		'quantity'  => 2,
		'attendees' => [
			0 => [
				'order_status' => 'yes',
				'full_name'    => 'Me Who',
				'email'        => 'me@who.com',
				'meta'         => [
					$field_slug => 'Attendee 1 field value',
				],
			],
			1 => [
				'full_name' => 'You Who',
				'email'     => 'you@who.com',
				'meta'      => [
					$field_slug => 'Attendee 2 field value',
				],
			],
		],
	],
];
```

[ET-858]: https://moderntribe.atlassian.net/browse/ET-858

### CODE REVIEW

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 1]
```diff
@@ -103,7 +103,10 @@ public function add_to_cart() {
 
 		// bail if this isn't a Tribe Commerce PayPal ticket
 		if (
-			empty( $_POST['product_id'] )
+			(
+				empty( $_POST['tribe_ticket_id'] )
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:107]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 8a47a36defa5d3a536bc38a541eb5a00</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 2]
```diff
@@ -103,7 +103,10 @@ public function add_to_cart() {
...
+				&& empty( $_POST['product_id'] )
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:108]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: d30d2e3bf74b0ea4e29c3dd7e300d6e2</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 3]
```diff
@@ -113,7 +116,12 @@ public function add_to_cart() {
 		$cart_url      = $this->get_cart_url( '_cart' );
 		$post_url      = get_permalink( $post );
 		$currency_code = trim( tribe_get_option( 'ticket-commerce-currency-code' ) );
-		$product_ids   = (array) $_POST['product_id'];
+
+		if ( isset( $_POST['tribe_ticket_id'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:120]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 36322951b406f9915032d8d8dc1fc94d</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 4]
```diff
@@ -113,7 +116,12 @@ public function add_to_cart() {
...
+			$product_ids = (array) $_POST['tribe_ticket_id'];
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:121]: * :warning: Detected usage of a non-sanitized input variable: $_POST['tribe_ticket_id']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 7404c006d9f1a2fefc0e744a96e7c806</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 5]
```diff
@@ -113,7 +116,12 @@ public function add_to_cart() {
...
+		} elseif ( isset( $_POST['product_id'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:122]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: d4e3dded51b94ecb388f95634c3432fb</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 6]
```diff
@@ -113,7 +116,12 @@ public function add_to_cart() {
...
+			$product_ids = (array) $_POST['product_id'];
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:123]: * :warning: Detected usage of a non-sanitized input variable: $_POST['product_id']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 54d835db2b57657137707949c08aa37c</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 7]
```diff
@@ -175,7 +183,14 @@ public function add_to_cart() {
 
 		foreach ( $product_ids as $ticket_id ) {
 			$ticket   = tribe( 'tickets.commerce.paypal' )->get_ticket( $post->ID, $ticket_id );
-			$quantity = absint( $_POST[ "quantity_{$ticket_id}" ] );
+
+			$quantity = 0;
+
+			if ( isset( $_POST['tribe_ticket_quantity'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:189]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: fab023424188a42b96312313198e0833</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 8]
```diff
@@ -175,7 +183,14 @@ public function add_to_cart() {
...
+				$quantity = absint( $_POST['tribe_ticket_quantity'] );
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:190]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: b5971c0b5fd83c2a8f557d1a368cbe6b</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 9]
```diff
@@ -175,7 +183,14 @@ public function add_to_cart() {
...
+			} elseif ( isset( $_POST[ "quantity_{$ticket_id}" ] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:191]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: f6609a235f305cbed9c273c3b2b3681b</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php [hunk 10]
```diff
@@ -175,7 +183,14 @@ public function add_to_cart() {
...
+				$quantity = absint( $_POST["quantity_{$ticket_id}"] );
```
Comment (tr1b0t) [inline src/Tribe/Commerce/PayPal/Gateway.php:192]: * :no_entry_sign: Processing form data without nonce verification.
* :no_entry_sign: Array keys must be surrounded by spaces unless they contain a string or an integer.
* :no_entry_sign: There must be a single space after the first square bracket
* :no_entry_sign: There must be a single space before the last square bracket

<sub>hash: 15a2e88450b95c278be7e642fc10b34d</sub>

File: src/Tribe/RSVP.php [hunk 11]
```diff
@@ -719,7 +725,17 @@ public function maybe_send_tickets_after_status_change( $event_id ) {
 	 */
 	public function generate_tickets() {
 
-		if ( empty( $_POST['tickets_process'] ) || empty( $_POST['attendee'] ) || empty( $_POST['product_id'] ) ) {
+		if (
+			empty( $_POST['tickets_process'] )
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:729]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 5bc5da7d72ca9b995286ef7e066deabe</sub>

File: src/Tribe/RSVP.php [hunk 12]
```diff
@@ -719,7 +725,17 @@ public function maybe_send_tickets_after_status_change( $event_id ) {
...
+			|| (
+				empty( $_POST['tribe_ticket_attendee'] )
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:731]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 55c28a60e6314f6e85d185f05441ede9</sub>

File: src/Tribe/RSVP.php [hunk 13]
```diff
@@ -719,7 +725,17 @@ public function maybe_send_tickets_after_status_change( $event_id ) {
...
+				&& empty( $_POST['attendee'] )
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:732]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 2d72fbecaa472150f2b0a10bb6f0b4cf</sub>

File: src/Tribe/RSVP.php [hunk 14]
```diff
@@ -719,7 +725,17 @@ public function maybe_send_tickets_after_status_change( $event_id ) {
...
+			)
+			|| (
+				empty( $_POST['tribe_ticket_id'] )
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:735]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: d9ffc1406ef2c46e13b60ccd74be7f99</sub>

File: src/Tribe/RSVP.php [hunk 15]
```diff
@@ -719,7 +725,17 @@ public function maybe_send_tickets_after_status_change( $event_id ) {
...
+				&& empty( $_POST['product_id'] )
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:736]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 0ed3a44094e42578220cdaa5a41f6010</sub>

File: src/Tribe/RSVP.php [hunk 16]
```diff
@@ -744,10 +760,20 @@ public function generate_tickets() {
 			tribe_exit();
 		}
 
-		// Iterate over each product
-		foreach ( (array) $_POST['product_id'] as $product_id ) {
-			if ( ! $ticket_qty = $this->parse_ticket_quantity( $product_id ) ) {
-				// if there were no RSVP tickets for the product added to the cart, continue
+		$product_ids = [];
+
+		if ( isset( $_POST['tribe_ticket_id'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:765]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 17329eeb80fc40b418fd551b166ddb0c</sub>

File: src/Tribe/RSVP.php [hunk 17]
```diff
@@ -744,10 +760,20 @@ public function generate_tickets() {
...
+			$product_ids = (array) $_POST['tribe_ticket_id'];
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:766]: * :warning: Detected usage of a non-sanitized input variable: $_POST['tribe_ticket_id']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 3bad937af472145e3d80e8a739fc9af6</sub>

File: src/Tribe/RSVP.php [hunk 18]
```diff
@@ -744,10 +760,20 @@ public function generate_tickets() {
...
+		} elseif ( isset( $_POST['product_id'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:767]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: db869f32765e87727706372a4fda74d2</sub>

File: src/Tribe/RSVP.php [hunk 19]
```diff
@@ -744,10 +760,20 @@ public function generate_tickets() {
...
+			$product_ids = (array) $_POST['product_id'];
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:768]: * :warning: Detected usage of a non-sanitized input variable: $_POST['product_id']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: eb3d9485f9281cc02225ab76bb7de36d</sub>

File: src/Tribe/RSVP.php [hunk 20]
```diff
@@ -2273,33 +2299,41 @@ public function generate_tickets_for( $product_id, $ticket_qty, $attendee_detail
 	public function parse_attendee_details() {
 		$order_id = self::generate_order_id();
 
-		$attendee_email     = empty( $_POST['attendee']['email'] ) ? null : sanitize_email( $_POST['attendee']['email'] );
+		$first_attendee = [];
+
+		if ( isset( $_POST['tribe_ticket_attendee'] ) && ! empty( $_POST['tribe_ticket_attendee'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2304]: * :no_entry_sign: Processing form data without nonce verification.
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 10e610b0331ed36014f472e10dfbe2a4</sub>

File: src/Tribe/RSVP.php [hunk 21]
```diff
@@ -2273,33 +2299,41 @@ public function generate_tickets_for( $product_id, $ticket_qty, $attendee_detail
...
+			$first_attendee = $_POST['tribe_ticket_attendee'][0];
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2305]: * :warning: Detected usage of a non-sanitized input variable: $_POST['tribe_ticket_attendee'][0]
* :no_entry_sign: Processing form data without nonce verification.
* :no_entry_sign: Detected usage of a possibly undefined superglobal array index: $_POST['tribe_ticket_attendee'][0]. Use isset() or empty() to check the index exists before using it

<sub>hash: feca720489537a165f3c7c540b074b76</sub>

File: src/Tribe/RSVP.php [hunk 22]
```diff
@@ -2273,33 +2299,41 @@ public function generate_tickets_for( $product_id, $ticket_qty, $attendee_detail
...
+		} elseif ( isset( $_POST['attendee'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2306]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 84d5f36d8d2e59a063a5c62bf92f95e7</sub>

File: src/Tribe/RSVP.php [hunk 23]
```diff
@@ -2273,33 +2299,41 @@ public function generate_tickets_for( $product_id, $ticket_qty, $attendee_detail
...
+			$first_attendee = $_POST['attendee'];
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2307]: * :warning: Detected usage of a non-sanitized input variable: $_POST['attendee']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: fe0135cb53090c261bd00a8defd41363</sub>

File: src/Tribe/RSVP.php [hunk 24]
```diff
@@ -2315,11 +2349,15 @@ public function parse_attendee_details() {
 	 *             any other case.
 	 */
 	public function parse_ticket_quantity( $product_id ) {
-		if ( empty( $_POST[ "quantity_{$product_id}" ] ) ) {
-			return 0;
+		$quantity = 0;
+
+		if ( isset( $_POST['tribe_ticket_quantity'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2354]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: ac368100ea88185fc51b72aa25784ab2</sub>

File: src/Tribe/RSVP.php [hunk 25]
```diff
@@ -2315,11 +2349,15 @@ public function parse_attendee_details() {
...
+			$quantity = absint( $_POST['tribe_ticket_quantity'] );
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2355]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 8225a0c5bece762d6794d420dae74d7c</sub>

File: src/Tribe/RSVP.php [hunk 26]
```diff
@@ -2315,11 +2349,15 @@ public function parse_attendee_details() {
...
+		} elseif ( isset( $_POST[ "quantity_{$ticket_id}" ] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2356]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: e28473165ebf56a321eb9b9be9bee7b0</sub>

File: src/Tribe/RSVP.php [hunk 27]
```diff
@@ -2315,11 +2349,15 @@ public function parse_attendee_details() {
...
+			$quantity = absint( $_POST["quantity_{$ticket_id}"] );
```
Comment (tr1b0t) [inline src/Tribe/RSVP.php:2357]: * :no_entry_sign: Processing form data without nonce verification.
* :no_entry_sign: Array keys must be surrounded by spaces unless they contain a string or an integer.
* :no_entry_sign: There must be a single space after the first square bracket
* :no_entry_sign: There must be a single space before the last square bracket

<sub>hash: b8899077bc48d80c024ca9df1b905787</sub>

File: src/Tribe/Tickets_View.php [hunk 28]
```diff
@@ -187,7 +187,8 @@ public function update_tickets() {
 		if (
 			empty( $_POST['process-tickets'] )
 			|| (
-				empty( $_POST['attendee'] )
+				empty( $_POST['tribe_ticket_attendee'] )
```
Comment (tr1b0t) [inline src/Tribe/Tickets_View.php:190]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: d8fcc7682a205b1cd2a147a8773af78a</sub>

File: src/Tribe/Tickets_View.php [hunk 29]
```diff
@@ -187,7 +187,8 @@ public function update_tickets() {
...
+				&& empty( $_POST['attendee'] )
```
Comment (tr1b0t) [inline src/Tribe/Tickets_View.php:191]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: ff939cd9b35e3566ce8c05c2cc8ef2ce</sub>

File: src/Tribe/Tickets_View.php [hunk 30]
```diff
@@ -196,7 +197,13 @@ public function update_tickets() {
 
 		$post_id = get_the_ID();
 
-		$attendees = ! empty( $_POST['attendee'] ) ? $_POST['attendee'] : [];
+		$attendees = [];
+
+		if ( isset( $_POST['tribe_ticket_attendee'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Tickets_View.php:202]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: d645a392d0c2144906e0cc1200a41547</sub>

File: src/Tribe/Tickets_View.php [hunk 31]
```diff
@@ -196,7 +197,13 @@ public function update_tickets() {
...
+			$attendees = $_POST['tribe_ticket_attendee'];
```
Comment (tr1b0t) [inline src/Tribe/Tickets_View.php:203]: * :warning: Detected usage of a non-sanitized input variable: $_POST['tribe_ticket_attendee']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: d17fdc5a30f5f90d3da73d64ab717ecc</sub>

File: src/Tribe/Tickets_View.php [hunk 32]
```diff
@@ -196,7 +197,13 @@ public function update_tickets() {
...
+		} elseif ( isset( $_POST['attendee'] ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Tickets_View.php:204]: :no_entry_sign: Processing form data without nonce verification.

<sub>hash: e3e44e4bb759ef0ec0418f8cdf1bfc5c</sub>

File: src/Tribe/Tickets_View.php [hunk 33]
```diff
@@ -196,7 +197,13 @@ public function update_tickets() {
...
+			$attendees = $_POST['attendee'];
```
Comment (tr1b0t) [inline src/Tribe/Tickets_View.php:205]: * :warning: Detected usage of a non-sanitized input variable: $_POST['attendee']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: 6d5d55d30bb79df17317cc745bd9c51d</sub>

File: src/Tribe/Commerce/PayPal/Gateway.php (hunk 10 above)
Comment (sc0ttkclark) [reply src/Tribe/Commerce/PayPal/Gateway.php:192]:
```suggestion
				$quantity = absint( $_POST[ "quantity_{$ticket_id}" ] );
```

File: src/views/v2/rsvp/form/fields/email.php [hunk 34]
```diff
@@ -35,7 +35,7 @@ class="tribe-common-b2--min-medium tribe-tickets__form-field-label"
 	<input
 		type="email"
 		class="tribe-common-form-control-text__input tribe-tickets__form-field-input tribe-tickets__rsvp-form-field-email"
-		name="attendee[email]"
+		name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][attendees][0][email]"
```
Comment (juanfra) [inline src/views/v2/rsvp/form/fields/email.php:38]: Let's fix the version TBD

File: src/views/v2/rsvp/form/fields/name.php [hunk 35]
```diff
@@ -35,10 +35,10 @@ class="tribe-common-b2--min-medium tribe-tickets__form-field-label"
 	<input
 		type="text"
 		class="tribe-common-form-control-text__input tribe-tickets__form-field-input tribe-tickets__rsvp-form-field-name"
-		name="attendee[full_name]"
+		name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][attendees][0][full_name]"
```
Comment (juanfra) [inline src/views/v2/rsvp/form/fields/name.php:38]: Let's fix the version TBD

File: src/views/v2/rsvp/form/fields/quantity.php [hunk 36]
```diff
@@ -26,8 +26,8 @@ class="tribe-common-b2--min-medium tribe-tickets__form-field-label"
 	</label>
 	<input
 		type="number"
-		name="quantity_<?php echo absint( $rsvp->ID ); ?>"
-		id="quantity_<?php echo absint( $rsvp->ID ); ?>"
+		name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][quantity]"
```
Comment (juanfra) [inline src/views/v2/rsvp/form/fields/quantity.php:29]: Let's fix the version TBD

**Pull request merged**